### ISSUE TITLE
BNGParser doesn't throw error if molecule in rule contains unspecified component

### ISSUE THREAD

**Issue by GoogleCodeExporter** [RuleWorld/bionetgen#67]
```
If the 'molecule types' block exists, inclusion of an unspecified component in 
a molecule in the 'reaction rules' block will cause BNG to throw the following 
error: "ERROR: Component t of molecule HSF(s,t,u,v) not found in molecule 
declaration." It would be nice if the BNGParser recognized this error as well.
```

Original issue reported on code.google.com by `lh64@cornell.edu` on 4 Dec 2012 at 11:47
